exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetReaderUtility.java

```diff
@@ -181,6 +186,101 @@ else if (parquetTableMetadata instanceof Metadata.ParquetTableMetadata_v2 &&
   }
 
   /**
+   * Checks that the metadata file has version less than
+   * the version where was changed the serialization of BINARY values
+   * and assigns byte arrays to min/max values obtained from the deserialized string.
+   *
+   * @param parquetTableMetadata table metadata that should be corrected
+   */
```

Comment:
This is very hard to follow for those of us not familiar with the details of the metadata file format. Please include a comment in the PR conversation that shows an example of the old and new formats. That will make it easy to double-check that the logic makes sense.